File: scala/com/kotori316/fluidtank/integration/top/TankDataProvider.java
package com.kotori316.fluidtank.integration.top;

import mcjty.theoneprobe.api.IProbeHitData;
import mcjty.theoneprobe.api.IProbeInfo;
import mcjty.theoneprobe.api.IProbeInfoProvider;
import mcjty.theoneprobe.api.ProbeMode;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.text.translation.I18n;
import net.minecraft.world.World;
import net.minecraftforge.fluids.FluidStack;

import com.kotori316.fluidtank.Config;
import com.kotori316.fluidtank.FluidTank;
import com.kotori316.fluidtank.Utils;
import com.kotori316.fluidtank.integration.Localize;
import com.kotori316.fluidtank.tiles.TileTankNoDisplay;

public class TankDataProvider implements IProbeInfoProvider {

    @Override
    public String getID() {
        return FluidTank.modID + ":toptank";
    }

    @SuppressWarnings("deprecation")
    @Override
    public void addProbeInfo(ProbeMode mode, IProbeInfo probeInfo, EntityPlayer player,
                             World world, IBlockState blockState, IProbeHitData data) {
        TileEntity entity = world.getTileEntity(data.getPos());
        if (entity instanceof TileTankNoDisplay && Config.content().showTOP()) {
            TileTankNoDisplay tank = (TileTankNoDisplay) entity;
            String tier = I18n.translateToLocalFormatted(Localize.WAILA_TIER, tank.tier().toString());
            String fluid = I18n.translateToLocalFormatted(Localize.WAILA_CONTENT,
                Utils.toJava(tank.connection().getFluidStack()).map(FluidStack::getLocalizedName).orElse("Null"));
            if (tank.connection().hasCreative()) {
                probeInfo.text(tier).text(fluid);
            } else {
                String amount = I18n.translateToLocalFormatted(Localize.WAILA_AMOUNT, tank.connection().amount());
                String capacity = I18n.translateToLocalFormatted(Localize.WAILA_CAPACITY, tank.connection().capacity());
                String comparator = I18n.translateToLocalFormatted(Localize.WAILA_COMPARATOR, tank.getComparatorLevel());
                probeInfo.text(tier).text(fluid).text(amount).text(capacity).text(comparator);
            }
        }
    }
}

File: scala/com/kotori316/fluidtank/integration/top/TankDataProvider.scala
package com.kotori316.fluidtank.integration.top

import com.kotori316.fluidtank.integration.Localize._
import com.kotori316.fluidtank.tiles.TileTankNoDisplay
import com.kotori316.fluidtank.{Config, FluidTank}
import mcjty.theoneprobe.api.{IProbeHitData, IProbeInfo, IProbeInfoProvider, ProbeMode}
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World

class TankDataProvider extends IProbeInfoProvider {
  override def getID = FluidTank.modID + ":toptank"

  //noinspection ScalaDeprecation
  override def addProbeInfo(mode: ProbeMode, probeInfo: IProbeInfo, player: EntityPlayer, world: World, blockState: IBlockState, data: IProbeHitData): Unit = {
    import net.minecraft.util.text.translation.I18n
    val entity = world.getTileEntity(data.getPos)
    entity match {
      case tank: TileTankNoDisplay if Config.content.showTOP =>
        val tier = I18n.translateToLocalFormatted(WAILA_TIER, tank.tier.toString)
        val fluid = I18n.translateToLocalFormatted(WAILA_CONTENT, tank.connection.getFluidStack.map(_.getLocalizedName).getOrElse(FLUID_NULL))
        val list = if (tank.connection.hasCreative) Seq(tier, fluid)
        else {
          val amount = I18n.translateToLocalFormatted(WAILA_AMOUNT, Long.box(tank.connection.amount))
          val capacity = I18n.translateToLocalFormatted(WAILA_CAPACITY, Long.box(tank.connection.capacity))
          val comparator = I18n.translateToLocalFormatted(WAILA_COMPARATOR, Int.box(tank.getComparatorLevel))
          Seq(tier, fluid, amount, capacity, comparator)
        }
        list.foreach(probeInfo.text)
      case _ =>
    }
  }
}